app/src/services/emailValidationService.js

```diff
@@ -0,0 +1,200 @@
+            sparkPostAPICalls: gladSparkpostCount,
+        };
+    }
+
+    static async getVIIRSEmailsValidationObject(date) {
+        const viirsSubs = await EmailValidationService.findVIIRSEmailSubscriptions(date);
+        const viirsSent = await EmailValidationService.countEmailsSentForSlugOnDate(date, 'viirs-active-fires');
+        const viirsSparkpostCount = await SparkpostService.getVIIRSCountInjectedOnDate(date);
+
+        return {
+            success: viirsSubs.length === viirsSent && viirsSent === viirsSparkpostCount,
+            expectedSubscriptionEmailsSent: viirsSubs.length,
+            actualSubscriptionEmailsSent: viirsSent,
+            sparkPostAPICalls: viirsSparkpostCount,
+        };
+    }
+
+    static async getMonthlyEmailsValidationObject(date) {
+        const monthlySubs = await EmailValidationService.findMonthlySummaryEmailSubscriptions(date);
+        const monthlySent = await EmailValidationService.countEmailsSentForSlugOnDate(date, 'monthly-summary');
+        const monthlySparkpostCount = await SparkpostService.getMonthlyCountInjectedOnDate(date);
+
+        return {
+            success: monthlySubs.length === monthlySent && monthlySent === monthlySparkpostCount,
+            expectedSubscriptionEmailsSent: monthlySubs.length,
+            actualSubscriptionEmailsSent: monthlySent,
+            sparkPostAPICalls: monthlySparkpostCount,
+        };
+    }
+
+    static getBeginAndEndDatesForCron(cron, date) {
+        const task = taskConfig.find((t) => t.dataset === cron);
+
+        const beginDate = date.clone()
+            .subtract(task.gap.value, task.gap.measure)
+            .subtract(task.periodicity.value, task.periodicity.measure)
+            .format('YYYY-MM-DD');
+
+        const endDate = date.clone()
+            .subtract(task.gap.value, task.gap.measure)
+            .format('YYYY-MM-DD');
+
+        return { beginDate, endDate };
+    }
+
+    static async hasGLADAlertsForDate(beginDate, endDate, params) {
+        const uri = await GLADAlertsService.getURLInPeriodForSubscription(beginDate, endDate, params);
+        const result = await ctRegisterMicroservice.requestToMicroservice({
+            uri: uri.replace('SELECT *', 'SELECT COUNT(*) as count'),
+            method: 'GET',
+            json: true,
+        });
+
+        return result.data && result.data[0] && result.data[0].count && result.data[0].count > 0;
+    }
+
+    static async hasVIIRSAlertsForDate(beginDate, endDate, params) {
+        const uri = await VIIRSAlertsService.getURLInPeriodForSubscription(beginDate, endDate, params);
+        const result = await ctRegisterMicroservice.requestToMicroservice({
+            uri: uri.replace('SELECT *', 'SELECT COUNT(*) as count'),
+            method: 'GET',
+            json: true,
+        });
+
+        return result.data && result.data[0] && result.data[0].count && result.data[0].count > 0;
+    }
+
+    static async findGLADEmailSubscriptions(date) {
```

Comment:
test something like:

`    static async findGLADEmailSubscriptions(date) {
        const subs = await Subscription.find({ 'resource.type': 'EMAIL', datasets: /glad-alerts/i });
        const { beginDate, endDate } = EmailValidationService.getBeginAndEndDatesForCron('glad-alerts', date);

        const subPromises = subs.map((sub) => EmailValidationService.hasGLADAlertsForDate(beginDate, endDate, sub.params));

        const subResults = await Promise.all(subPromises);

        return subResults.filter((e) => e === true).length;
    }`

Reply:
Much cleaner, thanks for the feedback :+1: